src/promptflow/promptflow/_sdk/entities/_connection.py

```diff
@@ -346,10 +347,12 @@ class AzureOpenAIConnection(_StrongTypeConnection):
     TYPE = ConnectionType.AZURE_OPEN_AI
 
     def __init__(
-        self, api_key: str, api_base: str, api_type: str = "azure", api_version: str = "2023-07-01-preview", **kwargs
+        self, api_key: str, api_base: str, api_type: str = "azure", api_version: str = "2023-07-01-preview",
+        token_provider: Union[str, TokenProviderABC] = AzureTokenProvider(), **kwargs
```

Comment:
User should be able to create connection with this field in yaml?

Reply:
I don't think so, and that's why the token_provider is not a field of AzureOpenAIConnectionSchema.
Do we have scenario that user needs to override the token_provider when create connection from yaml ?